### PR TITLE
Add assertions to ReminderCommandParser

### LINKED ISSUE #150: Insert Assertion Week 10

### PR DESCRIPTION
**Pull request by chewwh09**
- Add assertion in ReminderCommandParser class.
- Commented out some test classes due to assertion.

### CODE REVIEW

File: src/test/java/seedu/storemando/logic/parser/ReminderCommandParserTest.java [hunk 1]
```diff
@@ -25,11 +25,11 @@ public void parse_invalidSingleArg_throwsParseException() {
             String.format(MESSAGE_INVALID_COMMAND_FORMAT, ReminderCommand.MESSAGE_USAGE));
     }
 
-    @Test
-    public void parse_numberSmallerThanZero_throwsParseException() {
-        assertParseFailure(parser, "-1 day",
-            String.format(MESSAGE_INVALID_COMMAND_FORMAT, ReminderCommand.MESSAGE_USAGE));
-    }
+    //    @Test
+    //    public void parse_numberSmallerThanZero_throwsParseException() {
```
Comment (kumsssss) [inline src/test/java/seedu/storemando/logic/parser/ReminderCommandParserTest.java]: Can just delete the test. Don't need to comment it out

**Review by kumsssss — COMMENTED** [review]
LGTM!

File: src/main/java/seedu/storemando/logic/parser/ReminderCommandParser.java [hunk 2]
```diff
@@ -46,6 +46,7 @@ public ReminderCommand parse(String args) throws ParseException {
             String[] stringArgsArr = trimmedArgs.replaceAll("\\s{2,}", " ").split(" ");
             long parsedNumber = Long.parseLong(stringArgsArr[0]);
             String timeUnit = stringArgsArr[1];
+            assert parsedNumber > 0 : "Input number must be a positive number";
             long numOfDaysFromToday = timeConversion(parsedNumber, timeUnit);
 
             if (trimmedArgs.isEmpty() || numOfDaysFromToday <= 0) {
```
Comment (JayChenYJ) [inline src/main/java/seedu/storemando/logic/parser/ReminderCommandParser.java]: u did the assertion then here no need the if alr right :)

Comment (chewwh09) [reply src/main/java/seedu/storemando/logic/parser/ReminderCommandParser.java]: I made changes to the assertion. It's no longer in this class hence the if  clause is still valid.

File: src/test/java/seedu/storemando/logic/parser/ReminderCommandParserTest.java (hunk 1 above)
Comment (chewwh09) [reply src/test/java/seedu/storemando/logic/parser/ReminderCommandParserTest.java]: I added the test back.

**Review by JayChenYJ — APPROVED** [review]
LGTM

**Pull request merged**